### PR TITLE
STYLE: Re-factor vtkMRMLSliceCompositeNode to use vtkMRML macros

### PR DESCRIPTION
**Pull request by jcfr**
This PR integrates `vtkMRMLSliceCompositeNode` some of serialization implementation changes originally contributed in  [lassoan/SlicerGitSVNArchive@cleanup-view-widget-creation2](https://github.com/lassoan/SlicerGitSVNArchive/tree/cleanup-view-widget-creation2) and  [lassoan/SlicerGitSVNArchive@cleanup-view-widget-creation](https://github.com/lassoan/SlicerGitSVNArchive/tree/cleanup-view-widget-creation)

### CODE REVIEW

**Comment by jcfr** [comment]
This changes does not add any missing properties and all tests pass, additional commit will introduce fixes.

**Review by lassoan — COMMENTED** [review]
It looks good to me. If you touch this code it would be also nice to get rid of SetSceneReferences and UpdateReferences and use node references infrastructure to manage backgroundVolume, foregroundVolume, and labelVolume.

**Comment by jcfr** [comment]
> If you touch this code it would be also nice to get rid of SetSceneReferences and UpdateReferences and use node references infrastructure to manage backgroundVolume, foregroundVolume, and labelVolume.

Good point, I will do this in a follow up commit

File: Libs/MRML/Core/vtkMRMLSliceCompositeNode.cxx [hunk 1]
```diff
@@ -287,6 +287,9 @@ void vtkMRMLSliceCompositeNode::CopyContent(vtkMRMLNode* anode, bool deepCopy/*=
   vtkMRMLCopyIntMacro(FiducialVisibility);
   vtkMRMLCopyIntMacro(FiducialLabelVisibility);
   vtkMRMLCopyIntMacro(SliceIntersectionVisibility);
+  // To avoid breaking current implementation, copy of the "LayoutName" attribute
+  // will be enabled after revisiting the view initilization pipeline.
+  //vtkMRMLCopyStringMacro(LayoutName);
```
Comment (jcfr) [inline Libs/MRML/Core/vtkMRMLSliceCompositeNode.cxx:292]: @lassoan  I will submit the use of references for background/foreground/layer in `vtkMRMLSliceCompositeNode` in a different PR, in the meantime I would like to move foward with the integration of this one.

Is the comment I integrated above  sensible ?

**Review by lassoan — APPROVED** [review]
It is a step in the right direction, so it makes sense to merge this as is, even if further improvements are expected.

**Pull request merged**